### PR TITLE
(feat) - osx support with vmware fusion

### PR DESCRIPTION
**Pull request by eoinmcafee00**

### CODE REVIEW

**Review by tphoney — COMMENTED** [review]
you will need to rebase, and squash your commits

**Comment by eoinmcafee00** [comment]
@tphoney done

**Review by tphoney — CHANGES_REQUESTED** [review]

File: command/config/config.go [hunk 1]
```diff
@@ -70,10 +70,26 @@ type (
...
+		RootDirectory string `json:"root_directory,omitempty" yaml:"root_directory"`
+	}
+
 	// platform specifies the configuration for a platform instance.
 	platform struct {
-		OS   string `json:"os,omitempty"`
-		Arch string `json:"arch,omitempty"`
+		OS      string `json:"os,omitempty"`
+		Arch    string `json:"arch,omitempty"`
+		Version string `json:"version,omitempty"`
```
[the remark this answers is not in the document]
Comment (eoinmcafee00) [reply command/config/config.go:93]: Version is new, it's required for vmx config - it tell the VM which version of ISO is being used. I.e. darwin20 for bigSur

File: command/config/config.go [hunk 2]
```diff
@@ -70,10 +70,26 @@ type (
...
 	}
 
+	VMFusion struct {
+		Account struct {
+			Username string `json:"username,omitempty"  yaml:"username"`
+			Password string `json:"password,omitempty"  yaml:"password"`
+		}
+		ISO           string `json:"iso,omitempty"`
+		Name          string `json:"name,omitempty" yaml:"name"`
+		Memory        int64  `json:"memory,omitempty" yaml:"memory"`
```
[the remark this answers is not in the document]
Comment (eoinmcafee00) [reply command/config/config.go:81]: It does.

File: internal/drivers/vmfusion/pool.go [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+func (p *provider) GetMaxSize() int {
+	return p.limit
+}
+
+func (p *provider) GetMinSize() int {
+	return p.pool
+}
+
+func (p *provider) PingProvider(ctx context.Context) error {
+	return nil
```
[the remark this answers is not in the document]
Comment (eoinmcafee00) [reply internal/drivers/vmfusion/pool.go:55]: Good point, i had actually mean't to do that but missed it.

[the remark this answers is not in the document]
Comment (eoinmcafee00) [reply internal/drivers/vmfusion/pool.go:55]: I just checked, you don't need it to be running. "vmrun" works regardless

**Pull request merged**